bevy_editor_panes/bevy_scene_tree/src/lib.rs

```diff
@@ -57,122 +41,53 @@ fn setup_pane(pane: In<PaneStructure>, mut commands: Commands) {
-                        ..Default::default()
-                    },
-                    PickingBehavior::IGNORE,
-                )),
-            ))
-            .set_parent(scene_tree)
-            .observe(set_selected_entity_on_click)
-            .id();
-
-        commands
-            .entity(scene_entity)
-            .insert(HasSceneTreeRow(row_entity));
+        commands.entity(scene_tree).build_children(tree_rows);
     }
 }
 
-fn update_scene_tree_rows(
-    scene_query: Query<(Ref<Name>, &HasSceneTreeRow)>,
-    mut row_query: Query<(&SceneTreeRow, &mut BackgroundColor, Ref<PickingInteraction>)>,
-    children_query: Query<&Children>,
-    mut text_query: Query<&mut Text>,
-    selected_entity: Res<SelectedEntity>,
-) {
-    // Update row names if entity name changes
-    for (name, scene_tree_row) in &scene_query {
-        if name.is_changed() {
-            text_query
-                .get_mut(children_query.children(scene_tree_row.0)[0])
-                .unwrap()
-                .0 = name.deref().into();
-        }
-    }
-
-    // Update row color based on interaction state
-    for (scene_tree_row, mut background_color, picking_interaction) in &mut row_query {
-        let state = (
-            picking_interaction.deref(),
-            selected_entity.0 == Some(scene_tree_row.0),
-        );
-        background_color.0 = match state {
-            (_, true) => tailwind::NEUTRAL_700.into(),
-            (PickingInteraction::Hovered, _) => tailwind::NEUTRAL_500.into(),
-            _ => Color::NONE,
+fn scene_tree_row_for_entity(
+    entity: Entity,
+    name: &Name,
+    selected_entity: &SelectedEntity,
+) -> Template {
+    let set_selected_entity_on_click =
+        move |mut trigger: Trigger<Pointer<Click>>, mut selected_entity: ResMut<SelectedEntity>| {
+            if selected_entity.0 == Some(entity) {
+                selected_entity.0 = None;
+            } else {
+                selected_entity.0 = Some(entity);
+            }
+            trigger.propagate(false);
         };
-    }
-}
-
-/// Root UI node of the scene tree.
-#[derive(Component)]
-struct SceneTreeRoot;
 
-/// A scene tree row UI node.
-///
-/// Contains the row's corresponding scene entity.
-#[derive(Component)]
-struct SceneTreeRow(Entity);
-
-/// A component on an entity in the scene that holds the corresponding scene tree row UI node entity.
-///
-/// If the entity with this component is despawned, the corresponding scene tree row UI node entity
-/// will be automatically despawned.
-struct HasSceneTreeRow(Entity);
-
-impl Component for HasSceneTreeRow {
-    const STORAGE_TYPE: StorageType = StorageType::Table;
-
-    fn register_component_hooks(hooks: &mut ComponentHooks) {
-        hooks.on_remove(|mut world, entity, _| {
-            let row_entity = world.entity(entity).get::<Self>().unwrap().0;
-            world.commands().entity(row_entity).despawn_recursive();
-        });
+    template! {
+        {entity}: (
```

Comment:
It hadn't occurred to me that you would be able to use entities as anchors like this. Pleased to see that's not too complex.

Reply:
Yeah apparently entity's impl whatever trait you use for naming, which is great because otherwise the list order / hover state / etc would break when renaming an entity, if I went off of Name.